Pocket Casts Watch App/SourceInterfaceNavigationView.swift

```diff
@@ -0,0 +1,150 @@
+import SwiftUI
+import PocketCastsServer
+import PocketCastsUtils
+
+struct SourceRow: View {
+    let sourceSymbol: String
+    let label: String
+    let showPlusOnly: Bool
+    let active: Bool
+
+    var body: some View {
+        HStack {
+            Text(sourceSymbol)
+                .font(.title2)
+            VStack {
+                Text(label)
+                if showPlusOnly {
+                    HStack {
+                        Image("gold-plus")
+                        Image("plus-only")
+                    }
+                }
+            }
+            Spacer()
+            if active {
+                Image("now-playing-small")
+            }
+        }
+    }
+}
+
+struct UserRow: View {
+    let username: String
+    let profileImage: String
+    let isLoggedIn: Bool
+
+    var body: some View {
+        HStack {
+            Image(profileImage)
+            VStack(alignment: .leading) {
+                if isLoggedIn {
+                    Text(L10n.signedInAs)
+                        .font(.caption)
+                        .foregroundStyle(.gray)
+                }
+                Text(username)
+                    .multilineTextAlignment(.leading)
+            }
+        }
+    }
+}
+
+struct SourceInterfaceNavigationView: View {
+
+    @State var activeSource: Int? = SourceManager.shared.currentSource().rawValue
+
+    @StateObject var model = SourceInterfaceModel()
+
+    var body: some View {
+        NavigationView {
+            List {
+                Section {
+                    NavigationLink(destination: InterfaceView(source: .phone), tag: Source.phone.rawValue, selection: $activeSource) {
+                        SourceRow(sourceSymbol: L10n.phone.sourceUnicode(isWatch: false), label: L10n.phone, showPlusOnly: false, active: model.activeSource == .phone)
+                    }
+                    NavigationLink(destination: InterfaceView(source: .watch), tag: Source.watch.rawValue, selection: $activeSource) {
+                        SourceRow(sourceSymbol: L10n.watch.sourceUnicode(isWatch: true), label: L10n.watch, showPlusOnly: !model.isLoggedIn || !model.isPlusUser, active: model.activeSource == .watch)
+                    }.disabled(!model.isPlusUser)
+                } footer: {
+                    if model.isPlusUser {
+                        Text(L10n.watchSourceMsg)
+                            .font(.footnote)
+                            .multilineTextAlignment(.leading)
+                            .foregroundStyle(.gray)
+                    }
+                }
+                if model.isPlusUser {
+                    Section {
```

Comment:
I wonder if these Sections would be a little more readable as separate `ViewBuilder` functions here? I'm always torn whether we should pull these out but I think it could make sense here.

Reply:
Good suggestion, refactored here: https://github.com/Automattic/pocket-casts-ios/pull/1800/commits/07f98b5096c52547f7ebaf9f5c0516f583dd10c7